account_invoice_mode_at_shipping/models/stock_picking.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2020 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from odoo import models
+
+from odoo.addons.queue_job.job import job
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    def action_done(self):
+        res = super().action_done()
+
+        for picking in self:
+            if picking._invoice_at_shipping():
+                picking.with_delay()._invoicing_at_shipping()
+        return res
+
+    def _invoice_at_shipping(self):
+        """Check if picking must be invoiced at shipping."""
+        self.ensure_one()
+        return (
+            self.picking_type_code == "outgoing"
+            and self.sale_id.partner_invoice_id.invoicing_mode == "at_shipping"
+        )
+
+    @job(default_channel="root.invoice_at_shipping")
+    def _invoicing_at_shipping(self):
+        self.ensure_one()
+        sales_order = self._get_sales_order_to_invoice()
```

Comment:
I think you may need to filter sale order here if there are invocable lines, otherwise, it will raise an error 
https://github.com/odoo/odoo/blob/be6b59fb3bd52894524603ee81597d8ec289583a/addons/sale/models/sale.py#L629

Reply:
I have added a check for it, thanks.